jujugui/static/gui/src/app/components/lightbox/lightbox.js

```diff
@@ -1,35 +1,155 @@
-/* Copyright (C) 2015 Canonical Ltd. */
-
+/* Copyright (C) 2017 Canonical Ltd. */
 'use strict';
 
 const PropTypes = require('prop-types');
 const React = require('react');
+const classNames = require('classnames');
 
 const SvgIcon = require('../svg-icon/svg-icon');
 
 /**
   Renders a modal like overlay with a darkened background.
   A caption is displayed in white below the content if present.
 */
-class Lightbox extends React.PureComponent {
+class Lightbox extends React.Component {
+  constructor() {
+    super();
+
+    this.state = {
+      activeSlide: 0,
+      lastSlide: null
+    };
+  }
+
+  componentWillMount() {
+    if (this.props.children.map) {
+      this.setState({
+        lastSlide: this.props.children.length - 1
+      });
+    }
+  }
+
+  _goToSlide(index) {
+    if (index < 0) {
+      index = 0;
+    }
+    if (index > this.state.lastSlide) {
+      index = this.state.lastSlide;
+    }
+    this.setState({
+      activeSlide: index
+    });
+  }
+
+  _nextSlide() {
+    const nextSlide = this.state.activeSlide + 1;
+    if (nextSlide <= this.state.lastSlide) {
+      this._goToSlide(nextSlide);
+    }
+  }
+
+  _previousSlide() {
+    const nextSlide = this.state.activeSlide - 1;
+    if (nextSlide >= 0) {
+      this._goToSlide(nextSlide);
+    }
+  }
+
+  _generateNavigation() {
+    if (!this.state.lastSlide) {
+      return;
+    }
+
+    const bullets = [];
+
+    for(let i = 0, ii = this.state.lastSlide; i <= ii; i += 1) {
+      const classes = classNames(
+        'lightbox__navigation-bullet',
+        {
+          'is-active': this.state.activeSlide === i
+        }
+      );
+
+      bullets.push(
+        <li
+          className={classes}
+          key={i}
+          onClick={this._goToSlide.bind(this, i)}>&bull;</li>);
+    }
+
+    return (
+      <div className="lightbox__navigation">
+        <button
+          disabled={this.state.activeSlide === 0}
+          onClick={this._previousSlide.bind(this)}
+          className="lightbox__navigation-previous">
+          Previous
+        </button>
+        <button
+          disabled={this.state.activeSlide === this.state.lastSlide}
+          onClick={this._nextSlide.bind(this)}
+          className="lightbox__navigation-next">
+          Next
+        </button>
+        <ul className="lightbox__navigation-bullets">
+          {bullets}
+        </ul>
+      </div>
+    );
+  }
+
+  _generateContent() {
+    if (!this.props.children.map) {
```

Comment:
Is this to check if there are multiples?

Reply:
Yeah, I've updated it to use this.state.lastSlide instead.